src/server.c

```diff
@@ -6012,12 +6013,11 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
     /* Get info from modules.
      * if user asked for "everything" or "modules", or a specific section
      * that's not found yet. */
-    if (everything || dictFind(section_dict, "modules") != NULL || sections < (int)dictSize(section_dict)) {
-
+    if (everything || dictFind(section_dict, "modules") != NULL || sections_found < (int)dictSize(section_dict)) {
```

Comment:
maybe instead of all these changes, we can just let this code run if all_sections is set and the dict is non-empty?
i.e. the disadvantage vs your approach is that if someone does `INFO ALL commandstats` we'll also go search for module sections, but maybe that's not that bad.

the "optimization" of not going to look for module sections, will still kick in for `INFO ALL`, `INFO`, `INFO MEMORY`, `INFO MEMORY SERVER`, just not when both ALL and another non-module section is used.
but that's not a critical use case IMHO. the main case in which it'll be used AFAIK is `INFO ALL COMMANDSTATS`

Reply:
Sounds good to me.